### PR TITLE
UCP/TAG: Fix offload completion with inlined data

### PR DESCRIPTION
**Pull request by brminich**
## What
Fix bug introduced by #5889

### CODE REVIEW

File: src/ucp/tag/offload.c [hunk 1]
```diff
@@ -120,15 +120,13 @@ UCS_PROFILE_FUNC_VOID(ucp_tag_offload_completed,
...
     if (ucs_unlikely(inline_data != NULL)) {
         status = ucp_request_recv_data_unpack(req, inline_data, length, 0, 1);
-        ucs_mpool_put_inline(req->recv.tag.rdesc);
     } else if (req->recv.tag.rdesc != NULL) {
         status = ucp_request_recv_data_unpack(req, req->recv.tag.rdesc + 1,
                                               length, 0, 1);
-        ucs_mpool_put_inline(req->recv.tag.rdesc);
-    } else {
-        ucp_request_recv_buffer_dereg(req);
     }
```
Comment (yosefe) [inline src/ucp/tag/offload.c:126]:
```
if (inline_data) {
    ..
    ucp_tag_offload_release_buf
} else (rdesc != NULL) {
    put_inline
} else {
    dereg
}
```

File: test/gtest/ucp/test_ucp_tag_offload.cc [hunk 2]
```diff
@@ -311,6 +311,35 @@ UCS_TEST_P(test_ucp_tag_offload, connect)
...
+                                              send_callback);
+    if (UCS_PTR_IS_ERR(sreq)) {
+        ASSERT_UCS_OK(UCS_PTR_STATUS(sreq));
+    } else if (sreq != NULL) {
+        wait(sreq);
+        request_free(sreq);
+    }
+
+    wait(rreq);
+    request_free(rreq);
```
Comment (dmitrygx) [inline test/gtest/ucp/test_ucp_tag_offload.cc:340]: replace by `request_wait(rreq)` which releases request after completion

File: test/gtest/ucp/test_ucp_tag_offload.cc [hunk 3]
```diff
@@ -311,6 +311,35 @@ UCS_TEST_P(test_ucp_tag_offload, connect)
...
+    request *rreq = recv_nb_exp(&recvbuf[0], length, ucp_dt_make_contig(1), tag,
+                                UCP_TAG_MASK_FULL);
+
+    request *sreq = (request*)ucp_tag_send_nb(sender().ep(), &send_data, sizeof(send_data),
+                                              ucp_dt_make_contig(1), tag,
+                                              send_callback);
+    if (UCS_PTR_IS_ERR(sreq)) {
+        ASSERT_UCS_OK(UCS_PTR_STATUS(sreq));
+    } else if (sreq != NULL) {
+        wait(sreq);
```
Comment (dmitrygx) [inline test/gtest/ucp/test_ucp_tag_offload.cc:335]: replace by `request_wait(sreq)` which releases request after completion

File: src/ucp/tag/offload.c (hunk 1 above)
Comment (brminich) [reply src/ucp/tag/offload.c:130]: fixed

File: test/gtest/ucp/test_ucp_tag_offload.cc (hunk 2 above)
Comment (brminich) [reply test/gtest/ucp/test_ucp_tag_offload.cc]: fixed

File: test/gtest/ucp/test_ucp_tag_offload.cc (hunk 3 above)
Comment (brminich) [reply test/gtest/ucp/test_ucp_tag_offload.cc]: fixed

File: src/ucp/tag/offload.c [hunk 4]
```diff
@@ -129,6 +129,7 @@ UCS_PROFILE_FUNC_VOID(ucp_tag_offload_completed,
         ucp_request_recv_buffer_dereg(req);
     }
 
+
```
Comment (dmitrygx) [inline src/ucp/tag/offload.c:132]: excessive empty line

**Review by dmitrygx — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Comment by brminich** [comment]
the only failure is AZP static-check (due to docker image change), thus can merge

**Comment by brminich** [comment]
@yosefe, need to port to 1.10?

**Comment by yosefe** [comment]
Yes

**Pull request merged**